pennylane/interfaces/batch/torch.py

```diff
@@ -0,0 +1,236 @@
+
+        jac = torch.as_tensor(jac, device=device)
+        vjp = qml.gradients.compute_vjp(d, jac)
+        vjps.extend(vjp)
+
+    return vjps
+
+
+class ExecuteTapes(torch.autograd.Function):
+    """The signature of this ``torch.autograd.Function`` is designed to
+    work around Torch restrictions.
+
+    In particular, ``torch.autograd.Function``:
+
+    - Cannot accept keyword arguments. As a result, we pass a dictionary
+      as the first argument ``kwargs``. This dictionary **must** contain:
+
+      * ``"tapes"``: the quantum tapes to batch evaluate
+      * ``"device"``: the quantum device to use to evaluate the tapes
+      * ``"execute_fn"``: the execution function to use on forward passes
+      * ``"gradient_fn"``: the gradient transform function to use
+        for backward passes
+      * ``"gradient_kwargs"``: gradient keyword arguments to pass to the
+        gradient function
+      * ``"max_diff`"`: the maximum order of derivatives to support
+
+    Further, note that the ``parameters`` argument is dependent on the
+    ``tapes``; this Function should always be called
+    with the parameters extracted directly from the tapes as follows:
+
+    >>> parameters = []
+    >>> [parameters.extend(t.get_parameters()) for t in tapes]
+    >>> kwargs = {"tapes": tapes, "device": device, "gradient_fn": gradient_fn, ...}
+    >>> ExecuteTapes.apply(kwargs, *parameters)
+
+    The private argument ``_n`` is used to track nesting of derivatives, for example
+    if the nth-order derivative is requested. Do not set this argument unless you
+    understand the consequences!
+    """
+
+    @staticmethod
+    def forward(ctx, kwargs, *parameters):  # pylint: disable=arguments-differ
+        """Implements the forward pass batch tape evaluation."""
+        ctx.tapes = kwargs["tapes"]
+        ctx.device = kwargs["device"]
+
+        ctx.execute_fn = kwargs["execute_fn"]
+        ctx.gradient_fn = kwargs["gradient_fn"]
+
+        ctx.gradient_kwargs = kwargs["gradient_kwargs"]
+        ctx.max_diff = kwargs["max_diff"]
+        ctx._n = kwargs.get("_n", 1)
+
+        with qml.tape.Unwrap(*ctx.tapes):
+            res, ctx.jacs = ctx.execute_fn(ctx.tapes, **ctx.gradient_kwargs)
+
+        # if any input tensor uses the GPU, the output should as well
+        ctx.torch_device = None
+
+        for p in parameters:
+            if isinstance(p, torch.Tensor):
+                if p.is_cuda:  # pragma: no cover
+                    ctx.torch_device = p.get_device()
+                    break
+
+        for i, r in enumerate(res):
+            if r.dtype == np.dtype("object"):
+                # For backwards compatibility, we flatten ragged tape outputs
+                r = np.hstack(r)
+
+            res[i] = torch.as_tensor(torch.from_numpy(r), device=ctx.torch_device)
+
+            if ctx.jacs:
+                ctx.jacs[i] = torch.as_tensor(
+                    torch.from_numpy(ctx.jacs[i]), device=ctx.torch_device
+                )

```

Comment:
🤔 I'm tempted to leave it as is to favour code clarity here at the expense of overhead

The reason being is that I imagine that `if bool:` is probably one of the least significant conditionals for overhead, since nothing computationally heavy is happening!